### ISSUE TITLE
Slow, slow, slow !!!!

### ISSUE THREAD

**Issue by Mahabarata** [openstreetmap/iD#5971]
Hi,

For a few hours I was thinking my computer was out of order (or google chrome) : everytime I make a change, I need to wait 10 seconds because ID is frozen. When looking in the task manager, I can see that chrome uses 30% of the processor for 10 second then 0% : I go back to chrome, make a new change in ID and again 30% with chrome/ID frozen.
I have saved my changes, closed chrome, cleaned the cache, closed my computer and restart all : for 5 minutes looks great but then again chrome/ID frozen with almost no change in the last 5 minutes.
And then I saw that you have changed to a new version of ID a few hours ago.
This morning, same computer, same chrome but previous version of ID : no pb at all, I have made 176 improvements before to save my work and all was ok. Now, 10 changes and all is frozen.

Please do something quickly, It's impossible to work !

Best regards
Fred

**Comment by quincylvania** [comment]
@Mahabarata Sorry you're having such trouble! The new validation checks added in #5830 can definitely impact performance, especially if you're editing lots of long lines or areas. I can still get 100+ edits on my MacBook Pro on Safari without an appreciable slowdown. Can you share more details about your setup?

We're still working on ways to optimize validation, such as #5901. We'll also be adding controls to let you turn off some validations so they will not impact performance at all.

**Comment by quincylvania** [comment]
I just merged #5979 so in the next release of iD you'll be able to disable validation code in order to speed up the page. We'll also work in #5901 and similar optimizations so hopefully you won't have to turn anything off!

**Comment by quincylvania** [comment]
Closing this since it's not specifically actionable and #5901 covers an actionable task which will help quite a bit. @Mahabarata if you could post your setup details that would still be helpful!

**Comment by Mahabarata** [comment]
Sorry if I answer only now, I was on the road (again).
I've just arrived, I did't try ID since I opened this issue.
I think one of your suggestions of the problem (editing long lines) is the good one.
I have read the other issues you told me, I think you are on the good way.
Of course I will tell you if it is better or not.

My configuration :
Intel Pentium N3540 @ 2.16 Ghz
4 Go RAM
Windows 10 x64
google chrome 72

Best regards
And thanks for your help.

**Comment by Mahabarata** [comment]
Hi,

after a few days of tests, it looks quite better : sometimes it's still very slow but after a saving and sometimes to close/reopen chrome it works good.
In fact, when I opened this issue, the slowness appeared very quicky and I needed to save/restart chrome maybe every 15 minutes. Now I may work a lot, maybe 2 hours, before to have to do something.
So for me this problem is solved. Thanks again
Best regards
